File: Front/TS/src/infrastructure/adapters/presenters/CreateBetPresenter.ts
import { CreateBetResponse, type ICreateBetOutputPort } from '../../../domain/features/CreateBetHandler';
import { Presenter } from './Presenter';
export class CreateBetPresenter extends Presenter implements ICreateBetOutputPort {

    InvalidChip(): void {
        this.subjects.get(Key.CreateBetError.toString())?.forEach(x => x.next("Invalid chips"))
    }
    FriendsIsEmpty(): void {
        this.subjects.get(Key.CreateBetError.toString())?.forEach(x => x.next("One friend at least"))
    }
    DescriptionIsEmpty(): void {
        this.subjects.get(Key.CreateBetError.toString())?.forEach(x => x.next("Description is empty"))
    }
    InvalidEndDate(): void {
        this.subjects.get(Key.CreateBetError.toString())?.forEach(x => x.next("Invalid date"))
    }
    Present(createBetResponse: CreateBetResponse): void {
        this.subjects.get(Key.Success.toString())?.forEach(x => x.next(createBetResponse))
    }
}




export enum Key {
    CreateBetError,
    Success
}

File: Front/.Net Maui/BetFriends/Features/Bets/CreateBet/CreateBetPresenter.cs
using BetFriends.Domain.Features.CreateBet;
using BetFriends.Services;
using CommunityToolkit.Mvvm.Messaging;

namespace BetFriends.Features.Bets.CreateBet;

internal class CreateBetPresenter : ICreateBetOutputPort
{
    public void ChipsAreRequired()
    {
        WeakReferenceMessenger.Default.Send(new CreateBetException("Betting with no chip is forbidden"));
    }

    public void DescriptionIsEmpty()
    {
        WeakReferenceMessenger.Default.Send(new CreateBetException("Description is required"));
    }

    public void EndDateIsNotValide(DateTime endDate)
    {
        WeakReferenceMessenger.Default.Send(new CreateBetException($"{endDate.Date.ToShortDateString()} is not valide"));
    }

    public void FriendsIsRequired()
    {
        WeakReferenceMessenger.Default.Send(new CreateBetException("Need to bet with at least one friend"));
    }

    public void Present(CreateBetResponse createBetResponse)
    {
        Data.Chips -= createBetResponse.Chips;
        WeakReferenceMessenger.Default.Send(createBetResponse);
    }
}
